src/databricks/labs/ucx/workspace_access/generic.py

```diff
@@ -208,38 +217,72 @@ def _prepare_new_acl(
         return acl_requests
 
 
-class WorkspaceListing(Listing):
-    def __init__(self, ws: WorkspaceClient, num_threads=20, start_path: str | None = "/"):
-        super().__init__(..., ..., ...)
+class WorkspaceListing(Listing, CrawlerBase):
+    def __init__(
+        self,
+        ws: WorkspaceClient,
+        sql_backend: SqlBackend,
+        inventory_database: str,
+        num_threads=20,
+        start_path: str | None = "/",
+    ):
+        Listing.__init__(self, ..., ..., ...)
+        CrawlerBase.__init__(
+            self,
+            backend=sql_backend,
+            catalog="hive_metastore",
+            schema=inventory_database,
+            table="workspace_objects",
+            klass=WorkspaceObjectInfo,
+        )
         self._ws = ws
         self._num_threads = num_threads
         self._start_path = start_path
+        self._sql_backend = sql_backend
+        self._inventory_database = inventory_database
+
+    def _crawl(self) -> list[WorkspaceObjectInfo]:
+        return list(self._assess_workspace_listing())
+
+    def _assess_workspace_listing(self):
+        from databricks.labs.ucx.workspace_access.listing import WorkspaceListing
+
+        ws_listing = WorkspaceListing(self._ws, num_threads=self._num_threads, with_directories=False)
+        for _object in ws_listing.walk(self._start_path):
+            workspace_object_info = WorkspaceObjectInfo(
+                _object.object_type.name, str(_object.object_id), _object.path, _object.language
+            )
+            yield workspace_object_info
```

Comment:
Can you inline the variable? It's not used anywhere else

Reply:
done